File: senior/2019/soma.py
def count_rectangles(N, K, sequence):
    count = 0  # Contador de retângulos com soma igual a K

    for i in range(N):
        curr_sum = 0  # Variável para armazenar a soma do subarray

        for j in range(i, N):
            curr_sum += sequence[j]  # Adiciona o próximo elemento ao subarray

            if curr_sum == K:  # Se a soma do subarray é igual a K, incrementa o contador
                count += 1

    return count


# Leitura dos dados de entrada
N, K = map(int, input().split())
sequence = list(map(int, input().split()))

# Chamada da função para contar os retângulos
num_rectangles = count_rectangles(N, K, sequence)

# Impressão do resultado
print(num_rectangles)

File: senior/2019/c/soma.c
#include <stdio.h>

int count_rectangles(int N, int K, int sequence[]) {
    int count = 0;  // Contador de retângulos com soma igual a K

    for (int i = 0; i < N; i++) {
        int curr_sum = 0;  // Variável para armazenar a soma do subarray

        for (int j = i; j < N; j++) {
            curr_sum += sequence[j];  // Adiciona o próximo elemento ao subarray

            if (curr_sum == K) {  // Se a soma do subarray é igual a K, incrementa o contador
                count++;
            }
        }
    }

    return count;
}

int main() {
    int N, K;
    scanf("%d %d", &N, &K);

    int sequence[N];
    for (int i = 0; i < N; i++) {
        scanf("%d", &sequence[i]);
    }

    // Chamada da função para contar os retângulos
    int num_rectangles = count_rectangles(N, K, sequence);

    // Impressão do resultado
    printf("%d\n", num_rectangles);

    return 0;
}
